common/lib/kernel/vfio_device.cpp

```diff
@@ -247,6 +247,51 @@ bool Device::pciEnable() {
   return true;
 }
 
+int Device::platformInterruptInit(int efds[]){
+  size_t efd_cnt = 0; //FIXME: efds currently has a fixed size of 32. But we don't check that and there may be more efds.
+  for (auto irq : irqs) {
+    const size_t irqCount = irq.count;
+    const size_t irqSetSize =
+        sizeof(struct vfio_irq_set) + sizeof(int) * irqCount;
+
+    auto *irqSetBuf = new char[irqSetSize];
+    if (!irqSetBuf)
+      throw MemoryAllocationError();
+    auto *irqSet = reinterpret_cast<struct vfio_irq_set *>(irqSetBuf);
+
+    irqSet->argsz = irqSetSize;
+    // DATA_EVENTFD binds the interrupt to the provided eventfd.
+    // SET_ACTION_TRIGGER enables kernel->userspace signalling.
+    irqSet->flags = VFIO_IRQ_SET_DATA_EVENTFD | VFIO_IRQ_SET_ACTION_TRIGGER;
+    irqSet->index = irq.index;
+    irqSet->start = 0;
+    irqSet->count = irqCount;
+
+    // Now set the new eventfds
+    for (size_t i = 0; i < irqCount; i++) {
+      efds[efd_cnt+i] = eventfd(0, 0);
```

Comment:
Please add a boundary check for `sizeof(efds)`

Reply:
```suggestion
      if (i >= sizeof(efds/sizeof(efds[0]))) {
           throw RuntimeError("too many interrupts");
      }
      efds[efd_cnt+i] = eventfd(0, 0);
```